src/payment/strategies/digitalriver/digitalriver-payment-strategy.ts

```diff
@@ -0,0 +1,157 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType, NotInitializedError, NotInitializedErrorType } from '../../../common/error/errors';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import PaymentMethodActionCreator from '../../payment-method-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import PaymentStrategy from '../payment-strategy';
+
+import DigitalRiverJS, { DigitalRiverDropIn, DigitalRiverInitializeToken, OnCancelOrErrorResponse, OnReadyResponse, OnSuccessResponse } from './digitalriver';
+import DigitalRiverPaymentInitializeOptions from './digitalriver-payment-initialize-options';
+import DigitalRiverScriptLoader from './digitalriver-script-loader';
+
+export default class DigitalRiverPaymentStrategy implements PaymentStrategy {
+    private _digitalRiverJS?: DigitalRiverJS;
+    private _digitalRiverDropComponent?: DigitalRiverDropIn;
+    private _submitFormEvent?: () => void;
+    private _loadSuccessResponse?: OnSuccessResponse;
+    private _digitalRiverCheckoutId?: string;
+    private _digitalRiverInitializeOptions?: DigitalRiverPaymentInitializeOptions;
+
+    constructor(
+        private _store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _digitalRiverScriptLoader: DigitalRiverScriptLoader
+    ) {}
+
+    async initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        this._digitalRiverInitializeOptions = options.digitalriver;
+
+        const state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(options.methodId));
+        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
+        let clientToken: DigitalRiverInitializeToken;
+
+        if (!paymentMethod.clientToken) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        clientToken = JSON.parse(paymentMethod.clientToken);
+
+        const billing = state.billingAddress.getBillingAddressOrThrow();
+        const customer = state.customer.getCustomerOrThrow();
+        this._digitalRiverCheckoutId = clientToken.checkoutId;
+        this._submitFormEvent = this._getDigitalRiverInitializeOptions().onSubmitForm;
+
+        const configuration = {
+            sessionId: clientToken.sessionId,
+            options: { ...this._getDigitalRiverInitializeOptions().configuration },
+            billingAddress: {
+                firstName: billing.firstName,
+                lastName: billing.lastName,
+                email: billing.email || customer.email,
```

Comment:
is there any reason why you fallback into `customer.email` ?

`billing.email` should always mirror whatever is in `customer.email` if it's provided

Reply:
The reason is that `billing.email` is declared as optional in the `BillingAddress` interface and `customer.email` is required in the `Customer` interface.

`DigitalRiverDropInConfiguration` requires an email string as mandatory, another solution could be  `email: optional || '' ` but I think It doesn't look very nice.

I think we could move on with the current approach but let me know if would you prefer me to change that.